rn/Teacher/ios/Teacher/Submissions/QuizSubmissionList/ViewModel/QuizSubmissionListViewModel.swift

```diff
@@ -0,0 +1,99 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2023-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import Core
+import Combine
+
+class QuizSubmissionListViewModel: ObservableObject {
+    // MARK: - Outputs
+    @Published public private(set) var state: StoreState = .loading
+    @Published public private(set) var submissions: [QuizSubmissionListItemViewModel] = []
+    @Published public private(set) var filter: QuizSubmissionListFilter
+    @Published public var isShowingFilterSelector = false
+    @Published public var subTitle: String = ""
+    @Published public var showError: Bool = false
+    public let title = NSLocalizedString("Submissions", comment: "")
+    public let filters = QuizSubmissionListFilter.allCases
+
+    // MARK: - Inputs
+    public let refreshDidTrigger = PassthroughSubject<() -> Void, Never>()
+    public let messageUsersDidTap = PassthroughSubject<WeakViewController, Never>()
+    public let filterDidChange: CurrentValueSubject<QuizSubmissionListFilter, Never>
+
+    // MARK: - Private
+    private var subscriptions = Set<AnyCancellable>()
+    private let interactor: QuizSubmissionListInteractor
+
+    public init(router: Router, filterValue: QuizSubmissionListFilter, interactor: QuizSubmissionListInteractor) {
+        self.interactor = interactor
+        self.filter = filterValue
+        filterDidChange = CurrentValueSubject<QuizSubmissionListFilter, Never>(filterValue)
+        // MARK: - Output
```

Comment:
I would extract these to a function named like setupOutputBindings or so instead of these MARKs within the init method.